plugins/janus_videoroom.c

```diff
@@ -5095,61 +6239,29 @@ void janus_videoroom_setup_media(janus_plugin_session *handle) {
-			json_t *pub = json_object();
-			json_object_set_new(pub, "videoroom", json_string("event"));
-			json_object_set_new(pub, "room", string_ids ? json_string(participant->room_id_str) : json_integer(participant->room_id));
-			json_object_set_new(pub, "publishers", list);
-			if (participant->room) {
-				janus_mutex_lock(&participant->room->mutex);
-				janus_videoroom_notify_participants(participant, pub, FALSE);
-				janus_mutex_unlock(&participant->room->mutex);
-			}
-			json_decref(pub);
-			/* Also notify event handlers */
-			if(notify_events && gateway->events_is_enabled()) {
-				json_t *info = json_object();
-				json_object_set_new(info, "event", json_string("published"));
-				json_object_set_new(info, "room", string_ids ? json_string(participant->room_id_str) : json_integer(participant->room_id));
-				json_object_set_new(info, "id", string_ids ? json_string(participant->user_id_str) : json_integer(participant->user_id));
-				json_object_set_new(info, "audio_codec", json_string(janus_audiocodec_name(participant->acodec)));
-				json_object_set_new(info, "video_codec", json_string(janus_videocodec_name(participant->vcodec)));
-				gateway->notify_event(&janus_videoroom_plugin, session->handle, info);
-			}
+			janus_videoroom_notify_about_publisher(participant, FALSE);
 			janus_refcount_decrease(&participant->ref);
 		} else if(session->participant_type == janus_videoroom_p_type_subscriber) {
 			janus_videoroom_subscriber *s = (janus_videoroom_subscriber *)session->participant;
-			if(s && s->feed) {
-				janus_videoroom_publisher *p = s->feed;
-				if(p && p->session) {
-					janus_videoroom_reqpli(p, "New subscriber available");
-					/* Also notify event handlers */
-					if(notify_events && gateway->events_is_enabled()) {
-						json_t *info = json_object();
-						json_object_set_new(info, "event", json_string("subscribed"));
-						json_object_set_new(info, "room", string_ids ? json_string(p->room_id_str) : json_integer(p->room_id));
-						json_object_set_new(info, "feed", string_ids ? json_string(p->user_id_str) : json_integer(p->user_id));
-						gateway->notify_event(&janus_videoroom_plugin, session->handle, info);
+			if(s && s->streams) {
+				/* Send a PLI for all the video streams we subscribed to */
+				GList *temp = s->streams;
+				while(temp) {
+					janus_videoroom_subscriber_stream *ss = (janus_videoroom_subscriber_stream *)temp->data;
+					janus_videoroom_publisher_stream *ps = ss->publisher_streams ? ss->publisher_streams->data : NULL;
+					if(ps && ps->type == JANUS_VIDEOROOM_MEDIA_VIDEO && ps->publisher && ps->publisher->session) {
+						janus_videoroom_reqpli(ps, "New subscriber available");
 					}
+					temp = temp->next;
+				}
+				/* Also notify event handlers */
+				if(notify_events && gateway->events_is_enabled()) {
+					json_t *info = json_object();
+					json_object_set_new(info, "event", json_string("subscribed"));
+					json_object_set_new(info, "room", json_integer(s->room_id));
```

Comment:
@lminiero one tiny bug in this branch, the event from videoroom for `subscribed` is still using integer room id even though config use string id.

With string id config, I get something this:
```
{"emitter":"...","type":64,"timestamp":1618694821191801,"session_id":5564909303519302,"handle_id":337004046279573,"opaque_id":"videoroomtest-Byx47uLLjCCO","event":{"plugin":"janus.plugin.videoroom","data":{"event":"subscribed","room":0}}}
```

The room should be string, but I get `"room":0`.

Reply:
Ah good catch, thanks! I'll fix this later.